tiny_dnn/core/framework/tensor.h

```diff
@@ -312,16 +297,50 @@ auto host_data() {
     return *this;
   }
 
+  /**
+   * Fill tensor with another tensor repeated size times over first axis
+   * @tparam S storage of parameter tensor
+   * @param size
+   * @param tensor
+   * @return
+   */
+  template <typename S>
+  Tensor &fill(size_t size, Tensor<U, S> tensor) {
+    auto shape = tensor.shape();
+    shape.insert(shape.begin(), size);
+    storage_.reshape(shape);
+    for (size_t i = 0; i < size; ++i) storage_[i] = tensor.storage_;
+    return *this;
+  }
   // TODO(Randl): checked version
   /**
    * Reshape tensor
    * @param shape new shape
    */
   void reshape(const std::vector<size_t> &shape) { storage_.reshape(shape); }
 
+  /**
+   * Reshapes to shape {}
+   */
+  void clear() { storage_.reshape({}); }
```

Comment:
Since I haven't acquired memory manually, it should be released automatically too.